### ISSUE TITLE
Error when using tidy(conf.int = T) with glm.nb()

### ISSUE THREAD

**Issue by bakaburg1** [tidyverse/broom#241]
Hello,

If you use tidy(model, conf.int = T) with glm.nb() models it throws this error:

```
Error in data.frame(..., check.names = FALSE) : 
  arguments imply differing number of rows: 13, 16
```
The error is thrown only if conf.int is TRUE. Running just confint() on the model do work instead.

Regards,
Angelo

**Comment by strengejacke** [comment]
I think it depends on the model. For me, it worked fine with glm.nb-models. Have you tried to set the `effetcs`-argument to `fixed`?

```
broom::tidy(model, conf.int = T, effect = "fixed")
```

**Comment by bakaburg1** [comment]
Yes, I confirm that is a dataset specific problem. Using effects = "fixed" doesn't help.
Is there any more information I can provide?

**Comment by dmi3kno** [comment]
There seems to be bug in `broom::tidy` as it quietly drops `NAs`, but `broom::confint_tidy` does not
```

library(purrr)
library(dplyr)
library(tidyr)
mtcars$cv_chunk <- c(1L, 1L, 2L, 1L, 1L, 3L, 1L, 1L, 3L, 1L, 1L, 2L, 3L, 3L, 3L, 
                     2L, 2L, 1L, 1L, 2L, 1L, 3L, 2L, 2L, 1L, 2L, 2L, 1L, 1L, 2L, 2L, 
                     1L)

mtcars_model_list <- mtcars %>% 
  split(.$cv_chunk) %>% 
  map(~lm(mpg ~ cyl*qsec + gear - cv_chunk, data = .))

# We will just look at 3rd CV fold. 
# It has too few observations - no interaction term

# This fails
map(mtcars_model_list, ~broom::tidy(.x, conf.int=TRUE))[[3]]
>Error in data.frame(..., check.names = FALSE) : 
>  arguments imply differing number of rows: 4, 5

#This works, but note interaction term is missing
map(mtcars_model_list, ~broom::tidy(.x))[[3]]

# This produces NAs 
map(mtcars_model_list, ~broom::confint_tidy(.x))[[3]]

# so either `broom::tidy` should not drop NA or broom::confint_tidy should.
# i prefer the former
```

**Comment by strengejacke** [comment]
Might be related to https://github.com/tidyverse/broom/issues/166

**Comment by alexpghayes** [comment]
`confint_tidy` will drop all `NA` rows. Just for the record, dropping the all `NA` row in 

```
# This works, but note interaction term is missing
map(mtcars_model_list, ~broom::tidy(.x))[[3]]
```

is intended behavior. This should be fixed once I merge #345.
